fireant/tests/queries/test_build_joins.py

```diff
@@ -27,6 +30,46 @@ def test_dimension_with_join_includes_join_in_query(self):
                          'GROUP BY "$timestamp","$district-name" '
                          'ORDER BY "$timestamp","$district-name"', str(queries[0]))
 
+    def test_dimension_with_dataset_join_includes_a_sub_query_join_in_query(self):
```

Comment:
This test shows how everything is dynamically applied to both main and sub-queries. It includes filters too.